### PR TITLE
Add SimplePlanFragment to the SPI

### PR DESCRIPTION
**Pull request by rschlussel**
## Description
Add SimplePlanFragment to the SPI, as well as required dependencies (Partitioning, PartitioningScheme, StageExecutionDescriptor, PlanFragmeentId, PartititioningHandle).

Some of the  methods from Partitioning and PartitioningScheme were moved to a utility class PartitioningUtils.

## Motivation and Context
The SimplePlanFragment contains all the critical fields from PlanFragment, and will be used for plugin plancheckers on the entire PlanFragment (validate native plan).

## Impact
SPI Changes

## Test Plan
CI

## Contributor checklist

- [ ] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

SPI Changes
* Add ``Partitioning``, ``PartitioningScheme``, ``PartitioningHandle``, ``PlanFragmentId``, ``StageExecutionDescriptor`` and ``SimplePlanFragment`` to the SPI.
```

### CODE REVIEW

**Comment by steveburnett** [comment]
Just a nit to add the PR number to the release note entry.

```
== RELEASE NOTES ==

SPI Changes
* Add ``Partitioning``, ``PartitioningScheme``, ``PartitioningHandle``, ``PlanFragmentId``, ``StageExecutionDescriptor`` and ``SimplePlanFragment`` to the SPI. :pr:`23601`
```

**Comment by rschlussel** [comment]
> Just a nit to add the PR number to the release note entry.
> 
> ```
> == RELEASE NOTES ==
> 
> SPI Changes
> * Add ``Partitioning``, ``PartitioningScheme``, ``PartitioningHandle``, ``PlanFragmentId``, ``StageExecutionDescriptor`` and ``SimplePlanFragment`` to the SPI. :pr:`23601`
> ```

I thought Linsong did a new release of the release tools so this will happen automatically now.

File: presto-spi/src/main/java/com/facebook/presto/spi/plan/SimplePlanFragment.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        this.variables = requireNonNull(variables, "variables is null");
+        this.partitioning = requireNonNull(partitioning, "partitioning is null");
+        this.tableScanSchedulingOrder = unmodifiableList(requireNonNull(tableScanSchedulingOrder, "tableScanSchedulingOrder is null"));
+        this.stageExecutionDescriptor = requireNonNull(stageExecutionDescriptor, "stageExecutionDescriptor is null");
+        this.outputTableWriterFragment = outputTableWriterFragment;
+
+        checkArgument(new HashSet<>(root.getOutputVariables()).containsAll(partitioningScheme.getOutputLayout()),
+                format("Root node outputs (%s) does not include all fragment outputs (%s)", root.getOutputVariables(), partitioningScheme.getOutputLayout()));
+
+        this.partitioningScheme = requireNonNull(partitioningScheme, "partitioningScheme is null");
```
Comment (tdcmeehan) [inline presto-spi/src/main/java/com/facebook/presto/spi/plan/SimplePlanFragment.java:64]:
```suggestion
        requireNonNull(partitioningScheme, "partitioningScheme is null");
        checkArgument(new HashSet<>(root.getOutputVariables()).containsAll(partitioningScheme.getOutputLayout()),
                format("Root node outputs (%s) does not include all fragment outputs (%s)", root.getOutputVariables(), partitioningScheme.getOutputLayout()));

        this.partitioningScheme = partitioningScheme;
```

File: presto-spi/src/main/java/com/facebook/presto/spi/plan/SimplePlanFragment.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            @JsonProperty("tableScanSchedulingOrder") List<PlanNodeId> tableScanSchedulingOrder,
+            @JsonProperty("partitioningScheme") PartitioningScheme partitioningScheme,
+            @JsonProperty("stageExecutionDescriptor") StageExecutionDescriptor stageExecutionDescriptor,
+            @JsonProperty("outputTableWriterFragment") boolean outputTableWriterFragment)
+    {
+        this.id = requireNonNull(id, "id is null");
+        this.root = requireNonNull(root, "root is null");
+        this.variables = requireNonNull(variables, "variables is null");
+        this.partitioning = requireNonNull(partitioning, "partitioning is null");
+        this.tableScanSchedulingOrder = unmodifiableList(requireNonNull(tableScanSchedulingOrder, "tableScanSchedulingOrder is null"));
```
Comment (tdcmeehan) [inline presto-spi/src/main/java/com/facebook/presto/spi/plan/SimplePlanFragment.java:57]:
```suggestion
        requireNonNull(tableScanSchedulingOrder, "tableScanSchedulingOrder is null");
        this.tableScanSchedulingOrder = unmodifiableList(new ArrayList<>(tableScanSchedulingOrder));
```

**Review by tdcmeehan — COMMENTED** [review]
Just a couple of nits

**Review by BryanCutler — APPROVED** [review]
LGTM

**Review by tdcmeehan — APPROVED** [review]

**Comment by steveburnett** [comment]
> I thought Linsong did a new release of the release tools so this will happen automatically now.

I asked Linsong, and that hasn't happened yet.

**Pull request merged**